src/runtime.cc

```diff
@@ -96,6 +97,12 @@ namespace detail {
 		cfg = std::make_unique<config>(argc, argv, *default_logger);
 		graph_logger->set_level(cfg->get_log_level());
 
+		uint32 cores;
+		if((cores = affinity_cores_available()) < min_cores_needed) {
+			std::string err_msg = "NOT ENOUGH CORES. AVAILABLE:" + std::to_string(cores);
```

Comment:
Does this need to scream? ;)
(also there's no need to build the string in a cumbersome way here, you can use the fmt support of the logger)